libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -146,3 +151,22 @@ func (p *addHostMetadata) String() string {
 	return fmt.Sprintf("%v=[netinfo.enabled=[%v], cache.ttl=[%v]]",
 		processorName, p.config.NetInfoEnabled, p.config.CacheTTL)
 }
+
+func skipAddingHostMetadata(event *beat.Event) bool {
+	// If host fields exist(besides host.name added by libbeat) in event, skip add_host_metadata.
+	hostFields, err := event.Fields.GetValue("host")
+
+	// Don't skip if there are no fields
+	if err != nil || hostFields == nil {
+		return false
+	}
+
+	hostFieldsMap := hostFields.(common.MapStr)
```

Comment:
This will panic if the `host` is not a `common.MapStr`. Since Beats like Filebeat can handle arbitrary data it should be defensive (for example see https://github.com/elastic/beats/blob/95626b8f1690344312c0831ab2bdcbccffe4d089/metricbeat/mb/event.go#L207-L216).

Reply:
Thanks @andrewkroh for pointing this out! I will create a separate PR to fix this issue.

Reply:
This is the PR: https://github.com/elastic/beats/pull/20791